src/hawkbit-client.c

```diff
@@ -121,6 +121,34 @@ static size_t curl_write_to_file_cb(void *ptr, size_t size, size_t nmemb, struct
         return written;
 }
 
+/**
+ * @brief Add string to Curl headers, avoiding overwriting an existing
+ *        non-empty list on failure.
+ *
+ * @param[out] headers curl_slist** of already set headers
+ * @param[in]  string  Header to add
+ * @param[out] error   Error
+ * @return TRUE if string was added to headers successfully, FALSE otherwise (error set)
+ */
+static gboolean add_curl_header(struct curl_slist **headers, const char *string, GError **error)
+{
+        struct curl_slist *temp = NULL;
+
+        g_return_val_if_fail(string, FALSE);
+        g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+        temp = curl_slist_append(*headers, string);
+        if (!temp) {
+                curl_slist_free_all(*headers);
+                g_set_error(error, RHU_HAWKBIT_CLIENT_CURL_ERROR, -1, "Could not add header %s",
```

Comment:
Should we use -1 here, too?

Reply:
Let's use `CURLE_FAILED_INIT` for all those `-1` cases. Do you want to be able to distinguish these cases any further? I think that's not needed.

Reply:
Sounds sufficient, yes.